src/lib/validator.ts

```diff
@@ -79,6 +80,22 @@ class Validator {
     }
   }
 
+  validateUpdatedAt() {
+    if (!this.previous) {
+      return true;
+    }
+    if (typeof this.previous.updatedAt !== "number") {
+      errorMessage("This model's previous `updatedAt` is not a number");
```

Comment:
The `errorMessage` function call resulting in a thrown Error is a surprising side-effect that made this diff hard to reason about. I'd recommend something more like 
```js
throw createError('...')
```